eth/vm/forks/berlin/computation.py

```diff
@@ -7,7 +28,36 @@
 
 from .opcodes import BERLIN_OPCODES
 
-BERLIN_PRECOMPILES = MUIR_GLACIER_PRECOMPILES
+
+def _calculate_multiplication_complexity(base_length: int, modulus_length: int) -> int:
+    max_length = max(base_length, modulus_length)
+    words = math.ceil(max_length / 8)
+    return words**2
+
+
+@curry
+def _compute_modexp_gas_fee_eip_2565(data: bytes) -> int:
```

Comment:
Instead of currying here, you could just pass in `_compute_modexp_gas_fee_eip_2565` as the `gas_calculator` method
```suggestion
def _compute_modexp_gas_fee_eip_2565(data: bytes) -> int:
```